code/game/atoms.dm

```diff
@@ -718,6 +718,33 @@ its easier to just keep the beam vertical.
 		return TRUE //we applied blood to the item
 	return
 
+//this proc exists specifically for cases where the mob that originated the blood (aka the "donor") might not exist anymore, leading to bugs galore
+/atom/proc/add_blood_from_data(var/list/blood_data)
+	if (!( istype(blood_data) ))
+		return FALSE
+
+	if (!( src.flags ) & FPRINT)
+		return FALSE
+
+	if(!blood_DNA || !istype(blood_DNA, /list))	//if our list of DNA doesn't exist yet (or isn't a list) initialise it.
+		blood_DNA = list()
+
+	blood_color = blood_data["blood_colour"]
+
+	//adding blood to humans
+	if (istype(src, /mob/living/carbon/human))
```

Comment:
implement `/mob/living/carbon/human/add_blood_from_data` instead of typechecking src